cpp/lib/Authenticator.cpp

```diff
@@ -363,5 +360,178 @@ namespace Client
 
     return coded;
   }
+
+  AuthenticatorOKTA::AuthenticatorOKTA(
+      SF_CONNECT* connection) : m_connection(connection)
+  {
+      m_account = m_connection->account;
+      m_authenticator = m_connection->authenticator;
+      m_user = m_connection->user;
+      m_password = m_connection->password;
+      m_port = m_connection->port;
+      m_host = m_connection->host;
+      m_protocol = m_connection->protocol;
+      m_disableSamlUrlCheck = m_connection->disable_saml_url_check;
+      m_retriedCount = 0;
+      m_retryTimeout = get_retry_timeout(m_connection);
+
+      //m_appID = m_connection->application_name;
+      //m_appVersion = m_connection->application_version;
+      m_appID = "ODBC";
+      m_appVersion = "3.4.1";
+  }
+
+  AuthenticatorOKTA::~AuthenticatorOKTA()
+  {
+      // nop
+  }
+
+
+  bool AuthenticatorOKTA::curl_post_call(SFURL& url, const jsonObject_t& obj, jsonObject_t& resp)
+  {
+      bool ret = true;
+      std::string destination = url.toString();
+      void* curl_desc;
+      CURL* curl;
+      curl_desc = get_curl_desc_from_pool(destination.c_str(), m_connection->proxy, m_connection->no_proxy);
+      curl = get_curl_from_desc(curl_desc);
+      SF_ERROR_STRUCT* err = &m_connection->error;
+
+      int64 elapsedTime = 0;
+      int8 maxRetryCount = get_login_retry_count(m_connection);
+      int64 renewTimeout = auth_get_renew_timeout(m_connection);
+
+      // add headers for account and authentication
+      SF_HEADER* httpExtraHeaders = sf_header_create();
+      std::string s_body = value(obj).serialize();
+      cJSON* resp_data = NULL;
+      httpExtraHeaders->use_application_json_accept_type = SF_BOOLEAN_TRUE;
+      if (!create_header(m_connection, httpExtraHeaders, &m_connection->error)) {
+          CXX_LOG_TRACE("sf", "IDPAuthenticator",
+              "post_curl_call",
+              "Failed to create the header for the request to get the token URL and the SSO URL");
+          m_errMsg = "OktaConnectionFailed: failed to create the header";
+          ret = false;
+      }
+
+      if (ret)
+      {
+          if (!::curl_post_call(m_connection, curl, (char*)destination.c_str(), httpExtraHeaders, (char*)s_body.c_str(),
```

Comment:
Maybe rename the virtual functions with C++ like name such as curlPostCall, so you won't have naming conflict here.

Reply:
I renamed the function. Thank you.